### PR TITLE
fix debug rendering transforms

### PR DESCRIPTION
**Pull request by ZappoMan**
This fixes the issue @AndrewMeadows was reporting about the simulation owner debugging

### CODE REVIEW

File: libraries/entities-renderer/src/RenderableDebugableEntityItem.cpp [hunk 1]
```diff
@@ -34,10 +32,9 @@ void RenderableDebugableEntityItem::render(EntityItem* entity, RenderArgs* args)
     if (args->_debugFlags & RenderArgs::RENDER_DEBUG_SIMULATION_OWNERSHIP) {
         Q_ASSERT(args->_batch);
         gpu::Batch& batch = *args->_batch;
-        Transform transform = entity->getTransformToCenter();
-        transform.postScale(entity->getDimensions());
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/RenderableDebugableEntityItem.cpp]: I believe the postScale() is acting additively, and so you'd end up with extra scaling

File: libraries/entities-renderer/src/RenderableSphereEntityItem.cpp [hunk 2]
```diff
@@ -39,7 +39,7 @@ void RenderableSphereEntityItem::render(RenderArgs* args) {
     
     Q_ASSERT(args->_batch);
     gpu::Batch& batch = *args->_batch;
-    batch.setModelTransform(getTransformToCenter());
+    batch.setModelTransform(getTransform()); // use a transform with scale, rotation, registration point and translation
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/RenderableSphereEntityItem.cpp]: spheres were all drawing with 1 meter diameter instead of being properly scaled.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 93c0bd63e962bc6673371e00bc9a1555570617e6 available at: https://gist.github.com/34a1873e50445a568d55<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5102.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5102.dmg

**Comment by Atlante45** [comment]
Do not merge - @ZappoMan was sleep deprived and drank too much rum while making that change.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 55ede2b39882e8e05c64db113a64747e66ef2f42 available at: https://gist.github.com/34a1873e50445a568d55<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5102.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5102.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 47888b46716d6808645eeb1ae3a3a96735ff82f7 available at: https://gist.github.com/34a1873e50445a568d55<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5102.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5102.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a878559e0c4a88126c924abc805f236f0f7bb765 available at: https://gist.github.com/34a1873e50445a568d55<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5102.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5102.dmg

**Pull request merged**